TrainingCenter/models/Student.cs

```
@@ -0,0 +1,66 @@
+﻿namespace TrainingCenter.models;
+
+public class Student
+{
+    private string name;
+    private Dictionary<Course, int> courseGrades;
+
+    public Student(string name)
+    {
+        this.name = name;
+        courseGrades = new Dictionary<Course, int>();
+    }
+
+    public void AddCourse(Course course)
+    {
+        course.AddStudentToCourse(this);
+        courseGrades.Add(course, 0);
+    }
+
+    public IEnumerable<Course> GetCourses()
+    {
+        return courseGrades.Keys;
+    }
+
+    private void SetCourseGrade(Course course, int grade)
+    {
+        if (courseGrades.ContainsKey(course))
```

otherwise maybe you should throw an exception?